src/App.tsx

```diff
@@ -15,48 +21,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SORT_FIELD_NAME: SortType = SortType.Name;
+const SORT_FIELD_LENGTH: SortType = SortType.Length;
+
+function getPrepareGoods(goods: string[], sortField: SortType, reverse: boolean) {
+  let prepareGoods = [...goods];
+
+  if (sortField) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    prepareGoods = prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = React.useState(SortType.Defoult);
+  const [reverse, setReverse] = React.useState(false);
+  const renderGoods = getPrepareGoods(goodsFromServer, sortField, reverse);
+
+  function getReset() {
+    setReverse(false);
+    setSortField(SortType.Defoult);
+  }
+
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={sortField === SortType.Name
+            ? 'button is-info'
+            : 'button is-info is-light'}
```

Comment:
use [classnames](https://www.npmjs.com/package/classnames) library for managing dynamic classnames (here and in other cases)
alreade exists in project dependencies.